### PR TITLE
Fix binominal distribution

### PR DESCRIPTION
**Pull request by InfinityMod**
Hi, Pull Request #1807 ends in an infinite loop while running the test setup for the distributions, causing the tests to end until the time limit is reached. 

The misbehavior can be tested just by running the test_distributions setup or only especially the 
**test_vmapped_binominal_p0** function solely with the following code included directly after the imports in **test_distributions.py**:
```
# Enable 64bit support for higher accuracy
if sys.maxsize > 2**32:
    jax.config.update("jax_enable_x64", True)
```

While I can't 100% say why this endless loop only occurs when increasing the numeric accuracy to 64 bits, I'm sure to have a suitable solution for the behavior (which also seems faulty at 32 bits).

The problem stems from the **_binomial_dispatch** function in **util.py** (see the description commit). In short, zero values for the probability p are also passed to the dispatch function, even if filtered out afterward via _lax.cond_.  Therefore, the underlying while_loop in the **_binomial_inversion** function runs infinitely.

This merge request solves this issue by ensuring that no p values equal to zero are passed to the underlying functions. As _lax.cond_ is still filtering out the results of the zero-corrected values, there's no change for instances using the mentioned methods.

### CODE REVIEW

File: numpyro/distributions/util.py [hunk 1]
```diff
@@ -156,6 +156,8 @@ def _binom_inv_cond_fn(val):
 def _binomial_dispatch(key, p, n):
     def dispatch(key, p, n):
         is_le_mid = p <= 0.5
+        #Make sure p=0 is never taken into account as a fix for possible zeros in p.
```
Comment (fehiepsi) [inline numpyro/distributions/util.py:159]: could we just simply clip p to tiny? `jnp.clip(p, minval=jnp.finfo(p.dtype).tiny)`

Comment (InfinityMod) [reply numpyro/distributions/util.py:159]: This works for me: ```jnp.clip(p, jnp.finfo(jnp.float32).tiny)```

Comment (fehiepsi) [reply numpyro/distributions/util.py:159]: this might affect samples with small `p` in x64. How about updating `_binomial_inversion` to
```
geom = jnp.floor(jnp.log1p(-u) / log1_p) + 1
...
log1_p = jnp.log1p(-p)
log1_p = jnp.where(log1_p == 0, -jnp.finfo(log1_p.dtype).tiny, log1_p)
```
The issue seems to come from `log1_p=0` and `jnp.log1p(-u) < 0`, which leads to a negative `geom`

Comment (fehiepsi) [reply numpyro/distributions/util.py:159]: Could you also fix the lint issue? I think you need a space `# comment` for the comment.

**Review by fehiepsi — APPROVED** [review]
Thanks @InfinityMod! This issue is subtle.
